lib/install.js

```diff
@@ -382,11 +383,13 @@ Installer.prototype.loadAllDepsIntoIdealTree = function (cb) {
     steps.push([validateArgs, this.idealTree, this.args])
     steps.push([loadRequestedDeps, this.args, this.idealTree, saveDeps, cg.newGroup('loadRequestedDeps')])
   } else {
-    if (this.prod) {
+    if (this.prod && this.dev) {
```

Comment:
From your example, I'm a little curious about why we got this tree:

```
pkg@1.0.0 
├── libA@1.0.0
├── depA@1.2.7
```

Is this how we would resolve a similar conflict among transitive dependencies?

Would this:

```
pkg@1.0.0 
├─┬ libA@~1.0.0
│ ├── depA@~1.2.0
├─┬ libB@~1.0.0
│ ├── depA@^1.2.0
```

also yield the same tree (prefering a shared copy of `depA@1.2.7` instead of putting `depA@1.4.5` in `libB`)?